### PR TITLE
feat: v14

### PR DESCRIPTION
**Pull request by wolfy1339**
BREAKING CHANGE: `onUnhandledRequest` middleware option is removed
BREAKING CHANGE: Drop support for NodeJS v14, v16
BREAKING CHANGE: Replace support for Node.js http(s) Agents with documentation on using fetch dispatchers instead (via `@octokit/request`)
BREAKING CHANGE: Remove ability to pass custom request options, except from `method`, `headers`, `body`, `signal` (via `@octokit/request`)

### CODE REVIEW

**Comment by gr2m** [comment]
I wonder if we could remove [middlewares](https://github.com/octokit/app.js#createnodemiddlewareapp-options) from this release as part of the breaking change? The `createNodeMiddleware` function has node-specific logic, which actually goes counter our efforts to make Octokit work cross platforms. Same with the middlewares in `@octokit/oauth-app`: https://github.com/octokit/oauth-app.js#middlewares

I would move out the middlewares into their own packages, like `@octokit/middleware-node-app`, `@octokit/middleware-node-oauth-app`, `@octokit/middleware-aws-oauth-app`, etc.

It would be yet another "plugin" type, besides Octokit plugins and Octokit auth strategies.

Any thoughts?

**Review by gr2m — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -18,33 +18,31 @@
...
-    "@octokit/auth-unauthenticated": "^3.0.0",
-    "@octokit/core": "^4.0.0",
-    "@octokit/oauth-app": "^4.0.7",
-    "@octokit/plugin-paginate-rest": "^6.0.0",
-    "@octokit/types": "^9.0.0",
-    "@octokit/webhooks": "^10.0.0"
+    "@octokit/auth-app": "^5.0.4",
+    "@octokit/auth-unauthenticated": "^4.0.1",
+    "@octokit/core": "^5.0.0-beta.3",
+    "@octokit/oauth-app": "^5.0.0-beta.8",
```
Comment (gr2m) [inline package.json:24]: just fyi hold off updating to `^5.0.0` until we figure out https://github.com/octokit/core.js/pull/563#issuecomment-1599494689

**Comment by nickfloyd** [comment]
This should be good to go now! cc: @wolfy1339 @gr2m

**Comment by wolfy1339** [comment]
There's still some dependencies that need to be updated

**Comment by nickfloyd** [comment]
> 



> There's still some dependencies that need to be updated

Ok... I'll have a look at what's left.

**Comment by wolfy1339** [comment]
Only waiting on https://github.com/octokit/oauth-app.js/pull/436, then this will be ready

**Comment by wolfy1339** [comment]
Also, https://github.com/octokit/plugin-paginate-rest.js/pull/541

**Comment by wolfy1339** [comment]
This is good to go now. Let's ship this :rocket:

**Review by nickfloyd — APPROVED** [review]

**Review by oscard0m — APPROVED** [review]

**Pull request merged**